File: examples/hooks/python/post_hydrate_notice.py
"""Small example hook used by starter adapter profiles.

This script intentionally does something boring: it prints a JSON note that a hydrate step
completed. Real users can replace it with notifications, index refreshes, or environment-
specific automation once they opt in via a copied adapter profile.
"""

from __future__ import annotations

import json


def main() -> int:
    payload = {
        "event": "post-hydrate",
        "tool": "oac",
        "status": "ok",
        "note": "starter hook placeholder",
    }
    print(json.dumps(payload))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: examples/hooks/typescript/post_hydrate_notice.ts
/**
 * Small example hook used by starter adapter profiles.
 *
 * This script intentionally does something boring: it writes a JSON note that a hydrate step
 * completed. Real users can replace it with notifications, index refreshes, or environment-
 * specific automation once they opt in via a copied adapter profile.
 */

const payload = {
  event: "post-hydrate",
  tool: "oac",
  status: "ok",
  note: "typescript starter hook placeholder",
};

process.stdout.write(`${JSON.stringify(payload)}
`);
